x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/create/index.tsx

```diff
@@ -86,7 +87,17 @@ const CreateRulePageComponent: React.FC = () => {
   ] = useUserData();
   const { loading: listsConfigLoading, needsConfiguration: needsListsConfiguration } =
     useListsConfig();
-  const { navigateToApp } = useKibana().services.application;
+  const {
+    application: { navigateToApp },
+    executionContext,
+  } = useKibana().services;
+
+  // Application ID and current URL are traced automatically.
+  useExecutionContext(executionContext, {
+    page: SecurityPageName.rules,
+    id: 'create',
```

Comment:
I'd recommend putting this information into the page name. For example `${SecutiryPageName.rules}_create`.
The `id` should be reserved for things with high cardinality. I'm not super familiar with the security solution, but this is a field I would use to keep the `id` of a saved incident for example.

We should then be able to detect performance issues on a specific type or just ocunt the number of incidents worked on in an organization.